mlir/lib/Quantum/Utils/RemoveQuantum.cpp

```diff
@@ -59,6 +59,21 @@ void removeQuantumMeasurements(func::FuncOp &function, PatternRewriter &rewriter
     }
 }
 
+void replaceQuantumMeasurements(func::FuncOp &function, PatternRewriter &rewriter)
+{
+    function.walk([&](MeasurementProcess op) {
+        if (auto tensorType = dyn_cast<RankedTensorType>(op->getResults().front().getType())) {
+            auto shape = tensorType.getShape();
+            auto elemType = tensorType.getElementType();
+            rewriter.replaceOpWithNewOp<tensor::EmptyOp>(op, shape, elemType);
+        }
+        else {
+            rewriter.replaceOpWithNewOp<tensor::EmptyOp>(op, op->getResults().front().getType(),
+                                                         ValueRange{});
+        }
```

Comment:
I'm not sure this branch covers all cases. It looks like we assume the MeasurementProcess only has one result? `counts` for example has two results. The other thing I don't quite get is the then branch is for tensor results, but the else branch still generates a tensor value?

Reply:
Yes one value, we do not support counts in ZNE as well as in Gradients. For the second I will take a look

Reply:
The new branch on the scalar type looks, we probably just want to add a base case where the type is unrecognized, which could return a `failed` LogicalResult from the function, allowing the pass to abort gracefully.

Reply:
As discussed offline: just added op.emiterror otherwise I need to refactor multiple functions.